main/src/ca/uwaterloo/flix/Main.scala

```diff
@@ -361,6 +362,7 @@ object Main {
                      xprintphase: Set[String] = Set.empty,
                      xsummary: Boolean = false,
                      xparser: Boolean = false,
+                     xprinttyper: Option[String] = None,
```

Comment:
What's the reason not to dump _ALL_ type constraints?

Reply:
It would take forever :)

The current model outputs a file for _each iteration_ of constraint resolution, which means up to hundreds of files per definition.

Reply:
Uh oh. I wonder if that is not something best left tuning in code.

Reply:
The AST debugging is just dumping the ASTs.

Reply:
which is nice to get a general overview